src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -101,6 +106,43 @@ class RocksDbDataSource(
     }
   }
 
+  private def dbIterator: Resource[Task, RocksIterator] = {
+    Resource.fromAutoCloseable(Task(db.newIterator()))
+  }
+
+  private def namespaceIterator(namespace: Namespace): Resource[Task, RocksIterator] = {
+    Resource.fromAutoCloseable(Task(db.newIterator(handles(namespace))))
+  }
+
+  private def moveIterator(it: RocksIterator): Observable[Either[IterationError, (Array[Byte], Array[Byte])]] = {
+    Observable
+      .fromTask(Task(it.seekToFirst()))
+      .flatMap(_ => Observable.fromTask(Task(it.isValid)))
+      .filter(identity)
+      .flatMap { _ =>
+        Observable.fromTask(Task(Right(it.key(), it.value()))) ++ Observable
```

Comment:
What about changing order a bit?
I think that this could be simplified to form like:
```scala
Observable
  // I'm not a fan of this, but I don't know significantly better solution
  .fromTask(Task(it.seekToFirst())) 
  .flatMap(_ =>
    Observable.repeatEvalF(for { //Or unfold maybe?
      isValid <- Task(it.isValid)
      //This could use `Task.ifM`, but its readabilty is discussable
      item <- if (isValid) Task(Right(it.key(), it.value())) else Task(Left(IterationFinished))
      _ <- Task(it.next())
    } yield item))
  .takeWhile(_.isRight))
  .onErrorHandle(ex => Left(IterationError(ex))
```
So:
 1. there's no duplicated part with getting first item
 2. code structure is more flat

Reply:
so I like how this encoding gets rid of duplicated part, i will definitely go that way.

Although i will probably not use `.takeWhile(_.isRight))` and keep `Task.raiseError(IterationFinished)` part as i do not what `IterationFinished` to leak outside this function.

And i will probably not use `Task.ifM` , as in general i do not like it, and everytime i see it i need to check docs to be certain which case goes where.